doobie/src/main/scala/tofu/doobie/transactor/Txr.scala

```diff
@@ -0,0 +1,118 @@
+package tofu.doobie
+package transactor
+
+import cats.data.ReaderT
+import cats.{Monad, ~>}
+import doobie.{ConnectionIO, Transactor}
+import fs2.Stream
+import tofu.lift.Lift
+import tofu.syntax.funk._
+import tofu.{BracketThrow, HasContext}
+
+/**
+  * A simple facade for [[doobie.Transactor]] that holds an inner database effect type `DB[_]` and provides
+  * natural transformations from this effect to the target effect `F[_]`.
+  *
+  * The motivation for using this facade instead of `Transactor` is to:
+  * 1) initialize all its natural transformations early and remove the need for additional constraints
+  *    on `F[_]` later (e.g. `Bracket`);
+  * 2) be able to use another `DB[_]` effect besides `ConnectionIO` and build a layer of transactional logic
+  *    with it.
+  *
+  * N.B. The only extended DB effect provided below is `ConnectionRIO`. There are intentionally no effects
+  * based on `EitherT` to provide a separate error channel for business-errors. This is due to the (bad) design
+  * of Cats Effect. Doobie uses `Throwable`-bounded [[cats.effect.Resource]] and [[fs2.Stream]] under the hood
+  * for applying a [[doobie.util.transactor.Strategy]], and throwing an error through `EitherT`'s left channel
+  * does not lead to the proper `ExitCase` of a bracket. All errors on the DB layer must be thrown and handled
+  * via the `Throwable` channel of `ConnectionIO`.
+  */
+trait Txr[F[_]] {
+  type DB[_]
+
+  /** Interprets `DB` into `F`, applying the transactional strategy. */
+  def trans: DB ~> F
+
+  /** Interprets `DB` into `F`, no strategy applied. */
+  def rawTrans: DB ~> F
+
+  /** Translates the stream, applying the transactional strategy. */
+  def transP: Stream[DB, *] ~> Stream[F, *]
+
+  /** Translates the stream, no strategy applied. */
+  def rawTransP: Stream[DB, *] ~> Stream[F, *]
+}
+
+object Txr {
+  type Aux[F[_], DB0[_]]     = Txr[F] { type DB[x] = DB0[x] }
```

Comment:
@catostrophe , I'd propose to swap F[_] and DB0[_] type parameters here in order to give Aux[F, DB] more intuitive look as transactor essentially provides a set of transformations from DB[_] to F[_] (Aux[DB, F]).

Reply:
`Aux` pattern is widely used and the common way of defining an `Aux` alias is the following: type params go first, type members go last.
I don't think it's correct to swap the order just because there's `DB ~> F` transformation. What if I added the opposite one?

Reply:
@catostrophe, you mean as an alias like `type Trans[DB[_], F[_]] = Txr.Aux[F, DB]`?
Yes, I think this would be fine.

Reply:
No, I meant why to choose the order of params in `Aux` based on the direction of arrows? There might be two opposite arrows. And what then? See `cats.Parallel.Aux`.
I'm just saying that `Aux[F, DB0]` is the common way of defining an aux.

Reply:
What do you think of renaming `Txr` to `Trans` btw?

Reply:
@catostrophe, good idea, Trans is more intuitive name

Reply:
@oskin1 Hmm... I remembered why I chose a different name.

```scala
val trans  = Trans.contextual[F](transactor)
...
val res = dbProgram ||> trans.trans
```
I thought `trans.trans` might be confusing.

Compare with:
```scala
val txr  = Txr.contextual[F](transactor)
...
val res = dbProgram ||> txr.trans
```
TXN is a commonly used acronym for transaction. So I chose TXR for transactor.

Reply:
Ok, makes sense